src/components/dialog/dialog.test.js

```diff
@@ -0,0 +1,234 @@
+        <Button forwardRef={ref} onClick={() => setIsOpen(false)}>
+          This should be focused first now
+        </Button>
+
+        <Textbox label="Textbox1" value="Textbox1" />
+        <Textbox label="Textbox2" value="Textbox2" />
+        <Textbox label="Textbox3" value="Textbox3" />
+      </Dialog>
+    </>
+  );
+};
+
+context("Testing Dialog component", () => {
+  describe("should render Dialog component with props", () => {
+    it.each([[0], [1], [100], [1000]])(
+      "should render Dialog component with %s as a height parameter",
+      (height) => {
+        CypressMountWithProviders(<DialogComponent height={`${height}px`} />);
+
+        const { viewportHeight } = Cypress.config();
+
+        let resultHeight;
+        if (height >= viewportHeight - 20) {
+          resultHeight = viewportHeight - 20;
+        } else {
+          resultHeight = height;
+        }
+
+        dialogPreview()
+          .should("have.css", "height")
+          .and("contain", resultHeight);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should render Dialog using %s as a title",
+      (title) => {
+        CypressMountWithProviders(<DialogComponent title={title} />);
+
+        dialogTitle().should("have.text", title);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should render Dialog using %s as a subtitle",
+      (subtitle) => {
+        CypressMountWithProviders(
+          <DialogComponent title="Sample dialog" subtitle={subtitle} />
+        );
+
+        dialogSubtitle().should("have.text", subtitle);
+      }
+    );
+
+    it.each([
+      ["extra-small", 300],
+      ["small", 380],
+      ["medium-small", 540],
+      ["medium", 750],
+      ["medium-large", 850],
+      ["large", 960],
+      ["extra-large", 1080],
+    ])(
+      "should render Dialog component with %s as a size and has width property set to %s",
+      (size, width) => {
+        CypressMountWithProviders(<DialogComponent size={size} />);
+
+        dialogPreview()
+          .should("have.css", "width")
+          .and("contain", `${width}px`);
+      }
+    );
+
+    it("should not render close icon when ShowCloseIcon is set to false", () => {
+      CypressMountWithProviders(<DialogComponent showCloseIcon={false} />);
+
+      closeIconButton().should("not.exist");
+    });
+
+    it("should render close icon when ShowCloseIcon is set to true. When you click the CloseIcon the Dialog is closed", () => {
+      CypressMountWithProviders(<DialogComponent />);
+      dialogPreview().should("exist");
+      closeIconButton().click();
+      cy.wait(1000);
+      dialogPreview().should("not.exist");
+    });
+
+    it("should render Dialog with enabledEscKey prop and should be closed after clicking Escape button", () => {
+      CypressMountWithProviders(<DialogComponent />);
+      dialogPreview().should("exist");
+      dialogPreview().trigger("keyup", keyCode("Esc"));
```

Comment:
why `keyup` ?

Reply:
Hi Igor, Ed helped me with an issue with the tests (where I had to pass FocusFirstElement for them to pass). There was a problem in the way I was mounting the dialog. Once he fixed the mount, this test started to fail. Ed explained that I would need to use keyup here instead of keydown as the closeModal listener is listening for keyup events.